### PR TITLE
Secure Credentials support for Extenders

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

Secure Credentials functions and support for extenders to use this has been moved into zowe-explorer-api.  Extenders can consume this by use of the ProfilesCache.activateKeytarApis(initialized: boolean, isTheia: boolean).  Zowe Explorer is using this within the extension.ts file and all duplicate code has been removed from Zowe Explorer and now is within zowe-explorer-api.
example call from extender:
```
const log = imperative.Logger.getAppLogger();
const profiles = new ProfilesCache(log);
await profiles.activateKeytarApis(
            imperative.CredentialManagerFactory.initialized,
            EnvironmentManager.isTheia()
        );
```

NOTE: to be merged after https://github.com/zowe/vscode-extension-for-zowe/issues/1282

## Release Notes

Milestone: 1.15.0

Changelog: Secure Credentials support for Extenders to consume via ProfilesCache.activateKeytarApis(initialized: boolean, isTheia: boolean).

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/zowe-explorer-api/src/profiles/CoreUtils.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (lauren-li) [inline packages/zowe-explorer-api/src/profiles/CoreUtils.ts:1]: Maybe we could rename this file to `securityModuleUtils.ts`?

File: packages/zowe-explorer/src/utils/CoreUtils.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+/*
```
Comment (lauren-li) [inline packages/zowe-explorer/src/utils/CoreUtils.ts:1]: Maybe we could rename this file to `fsUtils.ts`?

**Review by lauren-li — APPROVED** [review]
Functionally, this seems to work well. I was able to start Zowe Explorer and use its profiles as before both with and without SCS. The same was true for using an extender that utilized the newly-available Keytar management functionality in the API (again both with and without SCS). I also tested and verified that this change does not break extenders that are currently using SCS by duplicating Zowe Explorer's SCS code.
All unit and expected integration tests are passing.

I made some suggestions for the renaming the `*/CoreUtils.ts` files, but am approving the PR as it otherwise looks good to me.

Thanks @JillieBeanSim for your hard work on this PR!

File: packages/zowe-explorer/src/utils/CoreUtils.ts (hunk 2 above)
Comment (phaumer) [reply packages/zowe-explorer/src/utils/CoreUtils.ts:1]: or maybe `TempFolder`? I am more a fan of defining classes with static members for these kind of utilities. That make the code easier to read and refactor.

**Review by phaumer — APPROVED** [review]
Thanks for this refactoring @JillieBeanSim. I think this PR is good to merge for the release as it works well. However, I would like to request another PR that 
1. Moves the `getSecurityModules()` into the `KeytarCredentialsManager`
2. Adds this call to the FTP extension to make it work with secure credentials as well and to demonstrate how to use the API.

File: packages/zowe-explorer/src/utils/CoreUtils.ts (hunk 2 above)
Comment (lauren-li) [reply packages/zowe-explorer/src/utils/CoreUtils.ts:1]: `TempFolder` works in my mind, too. I originally thought maybe the `cleanDir` function could be used outside of the temp folder context, but looking at the code more closely, it seems we aren't using it like that.

**Comment by JillieBeanSim** [comment]
issue created to address suggestions for code https://github.com/zowe/vscode-extension-for-zowe/issues/1316

**Comment by JillieBeanSim** [comment]
> Thanks for this refactoring @JillieBeanSim. I think this PR is good to merge for the release as it works well. However, I would like to request another PR that
> 
> 1. Moves the `getSecurityModules()` into the `KeytarCredentialsManager`
> 2. Adds this call to the FTP extension to make it work with secure credentials as well and to demonstrate how to use the API.

Thanks for the suggestions @phaumer .  With #2, the FTP extension seems to inherit the SCS functionality from Zowe Explorer.  I didn't see any difference testing with and without this call added to their code.  What we could do is meet up with the CICS plugin team and suggest using our new calls, the SCS and the Folder Creation, as I believe they have all of the duplicate code for SCS in their plugin and have a prerequisite on the CLI plugin being installed.  What do you think?

**Pull request merged**